include/pybind11/pytypes.h

```diff
@@ -1336,9 +1337,14 @@ class memoryview : public object {
         // Py_buffer uses signed sizes, strides and shape!..
         static std::vector<Py_ssize_t> py_strides { };
         static std::vector<Py_ssize_t> py_shape { };
+        static std::vector<std::string> formats = {
+            "?", "b", "B", "h", "H", "i", "I", "q", "Q", "f", "d", "g"
+        };
         buf.buf = info.ptr;
         buf.itemsize = info.itemsize;
-        buf.format = const_cast<char *>(info.format.c_str());
+        auto it = std::find(formats.begin(), formats.end(), info.format);
```

Comment:
I someone accidentally mistypes a format character, that error will be silently ignored here and the wrong buf.format (nullptr) assigned.

Reply:
What is the policy in pybind11 in this situation? Throwing an exception to handle this?